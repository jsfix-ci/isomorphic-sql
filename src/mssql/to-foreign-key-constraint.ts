export function toForeignKeyConstraint (table: string, column: string, targetTable: string, targetColumn: string) {
    return `ALTER TABLE ${table} ADD CONSTRAINT FK_${table}_${column}_${targetTable}_${targetColumn} 
FOREIGN KEY ${column} REFERENCES ${targetTable}(${targetColumn})`;
};