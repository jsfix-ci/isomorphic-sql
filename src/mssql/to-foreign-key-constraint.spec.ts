import { toForeignKeyConstraint } from './to-foreign-key-constraint';

describe('toForeignKeyConstraint', () => {
    it('returns an add-constraint expression for foreign keys', () => {
        expect(toForeignKeyConstraint('posts', 'author_id', 'author', 'id')).toMatchSnapshot();
    });
});