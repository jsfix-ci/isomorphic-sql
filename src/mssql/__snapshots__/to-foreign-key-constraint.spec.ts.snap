// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toForeignKeyConstraint returns an add-constraint expression for foreign keys 1`] = `
"ALTER TABLE posts ADD CONSTRAINT FK_posts_author_id_author_id 
FOREIGN KEY author_id REFERENCES author(id)"
`;
